### ISSUE TITLE
Unable to recreate Fintabnet TATR Model Performance

### ISSUE THREAD

**Issue by nandha1nks** [microsoft/table-transformer#149]
I attempted in recreating the TATR model on FinTabNet data refering to the [Aligning benchmark datasets for TSR](https://arxiv.org/pdf/2303.00716.pdf) 

I used the [same code](https://github.com/microsoft/table-transformer/blob/main/scripts/process_fintabnet.py) as suggested in the paper to create FinTabNet.a6 data.
I kept the number of epochs as 26 as mentioned in the paper, and used [structure_config.json](https://github.com/microsoft/table-transformer/blob/main/src/structure_config.json) as the config file during training. 

But the results I got after training the model from scratch and the [published model](https://huggingface.co/bsmock/TATR-v1.1-Fin) are not similar. 

Can you please help me understand if I am missing something? Sharing the results for comparison. 

## Fintabnet TATR Results
<img src="https://github.com/microsoft/table-transformer/assets/46973715/5075d061-be9b-496a-a5b5-50dc3dd12713" height="300"/>

## Retrained Model Results
<img src="https://github.com/microsoft/table-transformer/assets/46973715/72c5ce7c-0257-4dfd-bfb0-012bf99b2c1d" height="300"/>

**Comment by karthikgali** [comment]
@bsmock Could you please look into it.

**Comment by MohamedDonia** [comment]
@nandha1nks I have the same issue here, I trained structure model from scratch with default configurations and got lower results than the published one, @bsmock  is there any change in the default structure configuration file?

**Comment by bsmock** [comment]
TLDR: you likely need to train the model for longer.

In that paper specifically we mention that we standardize one epoch to mean 720,000 training samples. This is so an epoch has the same meaning whether we are training with PubTables-1M or FinTabNet.

The version of FinTabNet we train with, called FinTabNet.c, has only 78,536 training samples. What we want is to drop the learning rate by 0.9 every epoch = 720,000 samples, not every 78,536 samples. I believe we had to create a custom version of the code to do this. But for simplicity you can come very close to doing this with the current code by setting the “lr_drop” parameter to 9 when you train. This will drop the learning rate every 9 * 78,536 = 706,824 samples. Then you want to train for 9 * 30 = 270 epochs, because an epoch according to the training code is 78,536 samples for this dataset. I recommend also setting “checkpoint_freq” to 9 so you have 30 model checkpoints instead of 270 model checkpoints.

If you want to do *exactly* what we did in the paper, you might have to change the training code a little bit. But we released the model anyway for reproducibility.

Hope that helps!

**Comment by nandha1nks** [comment]
Thanks for clarifying that, @bsmock. Will try and reach out to you in case of other issue.
